aggregate rbac for crds to default roles

TheKangaroo opened the pull request:
this change adds ClusterRoles to aggregate rbac for
clustertriggerauthentications, scaledjobs, scaledobjects and
triggerauthentications to the kubernetes default roles view, edit and
admin via aggregation.
These ClusterRoles are optional and disabled by default. Set
rbac.aggregateToDefaultRoles to enable.



Signed-off-by: Till Adam <lxmail@web.de>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*

Fixes #256

JorTurFer commented on keda/Chart.yaml:
```diff
@@ -8,7 +8,7 @@ kubeVersion: ">=v1.17.0-0"
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 2.6.2
+version: 2.6.3
```
Please, revert this change because we bump the version during the chart release

JorTurFer reviewed (changes requested):
LGTM!
Thanks for this improvement! 
Only one small nit

JorTurFer commented:
@kedacore/keda-helm-contributors

TheKangaroo commented:
Ah thank you @JorTurFer, I reverted the version change.

tomkerkhove reviewed (approved):
LGTM, but be careful. We depend on what KEDA core generates as well so we might want to do it there as well.

TheKangaroo commented:
@tomkerkhove I'm not sure how keda core and helm chart plays together here, but if a PR in keda core is necessary, could you point me to the right direction?

tomkerkhove commented:
KEDA core generates YALM files to deploy but let's ask @zroubalik if this can be included first or if we should maintain it seperately

zroubalik commented:
Yeah, this should be changed in the core first. rbac is generated with kubebuilder from annotation on controllers, for example here: https://github.com/kedacore/keda/blob/9dbde78e238591870018a4eb9e2425b412ad26eb/controllers/keda/scaledobject_controller.go#L55-L63

tomkerkhove commented:
Are you up for contributing this in KEDA core as well @TheKangaroo?

TheKangaroo commented:
@tomkerkhove, sorry, this got lost on my side.
I had a look at @zroubalik's example and the kubebuilder docs, but I'm not sure how I can add aggregation roles via marker comments (i.e. add 'rbac.authorization.k8s.io/aggregate-to-xxxxx: "true"' labels to the generated clusterroles).

Or is ist just a matter of adding these clusterroles without any aggreation or bindings to the kedacore code? In this case, wouldn't it be better to drop the use of aggreation in this PR as well and just create clusterroles and clusterrolebindings instead?
Sorry if these are silly questions, I never worked with kubebuilder before.

tomkerkhove commented:
I will leave that up to @zroubalik to help though!

No worries, we are here to help you!

JorTurFer commented:
We should give some attention to this @kedacore/keda-helm-contributors

TheKangaroo commented:
I had a brief chat with @zroubalik via slack about the kubebuilder part, but we didn't manage to find time to do this yet.
We can give this another shot if one of you can help me with some hints on how to do this.

JorTurFer commented:
We are waiting because kubebuiler doesn't support this and we generate the rbac files using it, but considering that the changes are in metadata section, I guess that can make then here for helm only. Wdyt @zroubalik @TheKangaroo ?

TheKangaroo commented:
I just rebased the PR.
We switched to the AKS managed keda in the meantime, so we do not use this chart anymore, but I still think this is useful for others as well.

u0078867 commented:
Any news? Also interested here ...

zroubalik commented:
> As this is a role on top of other roles for giving access to KEDA resources as part of other cluster existing roles, I'd merge it without any change in core. @tomkerkhove @zroubalik ?

agree!

zroubalik reviewed (approved):
LGTM, thanks for the contribution!
